tests/test_baker.py

```diff
@@ -405,6 +405,43 @@ def test_regresstion_many_to_many_field_is_accepted_as_kwargs(self):
         assert store.customers.count() == 3
         assert models.Person.objects.count() == 6
 
+    def test_create_many_to_many_with_iter(self):
+        students = baker.make(models.Person, _quantity=3)
+        classrooms = baker.make(models.Classroom, _quantity=3, students=iter(students))
+
+        assert classrooms[0].students.count() == 1
+        assert classrooms[0].students.first() == students[0]
+        assert classrooms[1].students.count() == 1
+        assert classrooms[1].students.first() == students[1]
+        assert classrooms[2].students.count() == 1
+        assert classrooms[2].students.first() == students[2]
+
+    def test_create_many_to_many_with_unsaved_iter(self):
+        students = baker.prepare(models.Person, _quantity=3)
```

Comment:
Should this test also assert that the `students` are persisted once added to `classrooms`?

Reply:
When the `students` are added to the M2M they must be persistent instances, otherwise you hit this exception from the ORM:

<img width="1122" alt="image" src="https://user-images.githubusercontent.com/9328186/135298984-7dce9cf5-396c-4c1e-956b-490205a86c30.png">

Although this error isn't the greatest at explaining what's wrong IMO 🙂 

I added `students[...].pk is not None` checks in 294ae92 to clarify that the test expects the students to persist, however if the students were to not persist the test would actually hit the exception shown in my screenshot before hitting the primary key assertions